normandy/base/storage.py

```diff
@@ -6,20 +6,21 @@
 
 from inmemorystorage import InMemoryStorage
 from storages.backends.s3boto3 import S3Boto3Storage
+from storages.backends.gcloud import GoogleCloudStorage
 
 
 class PermissiveFilenameStorageMixin(object):
     """
     A storage class mixin that is is more permissive about valid filenames.
 
     In the original implemtnation, "anything that is not a unicode
-    alphanumeric, dash, underscore, or dot is removed". This version
-    allows most characters, except a few special characters. Leading
-    and trailing whitespaces are removed and internal whitespace is
-    converted to underscores.
+    alphanumeric, dash, underscore, or dot is removed". This modified version
+    allows most characters, except a few special characters. Leading and
+    trailing whitespaces are removed and internal whitespace is converted to
+    underscores.
 
-    Characters that S3 describes as requiring "significant special
-    handling for consistency across all applications" are removed.
+    File names produced follow recommended guidelines for both AWS S3 and GCP
+    Cloud Storage object names.
```

Comment:
nit: If you have a URL to that "recommended guidelines" that would be great.

Reply:
The URL is linked in the code, next to the actual changes to follow the guidelines. Does that work?